app/src/androidTest/java/com/duckduckgo/app/bookmarks/service/SavedSitesParserTest.kt

```diff
@@ -0,0 +1,222 @@
+        parser = RealSavedSitesParser()
+    }
+
+    @Test
+    fun whenSomeBookmarksExistThenHtmlIsGenerated() = runBlocking {
+        val bookmark = BookmarkEntity(id = 1, title = "example", url = "www.example.com")
+        val favorite = FavoriteEntity(id = 1, title = "example", url = "www.example.com", 0)
+
+        val result = parser.generateHtml(listOf(bookmark), listOf(favorite))
+        val expectedHtml = "<!DOCTYPE NETSCAPE-Bookmark-file-1>\n" +
+            "<!--This is an automatically generated file.\n" +
+            "It will be read and overwritten.\n" +
+            "Do Not Edit! -->\n" +
+            "<META HTTP-EQUIV=\"Content-Type\" CONTENT=\"text/html; charset=UTF-8\">\n" +
+            "<Title>Bookmarks</Title>\n" +
+            "<H1>Bookmarks</H1>\n" +
+            "<DL><p>\n" +
+            "    <DT><H3 ADD_DATE=\"1618844074\" LAST_MODIFIED=\"1618844074\" PERSONAL_TOOLBAR_FOLDER=\"true\">DuckDuckGo Bookmarks</H3>\n" +
+            "    <DL><p>\n" +
+            "        <DT><A HREF=\"www.example.com\" ADD_DATE=\"1618844074\" LAST_MODIFIED=\"1618844074\">example</A>\n" +
+            "    </DL><p>\n" +
+            "    <DT><H3 ADD_DATE=\"1618844074\" LAST_MODIFIED=\"1618844074\">DuckDuckGo Favorites</H3>\n" +
+            "    <DL><p>\n" +
+            "        <DT><A HREF=\"www.example.com\" ADD_DATE=\"1618844074\" LAST_MODIFIED=\"1618844074\">example</A>\n" +
+            "    </DL><p>\n" +
+            "</DL><p>\n"
+
+        assertEquals(expectedHtml, result)
+    }
+
+    @Test
+    fun whenNoSavedSitesExistThenNothingIsGenerated() = runBlocking {
+        val result = parser.generateHtml(emptyList(), emptyList())
+        val expectedHtml = ""
+
+        assertEquals(expectedHtml, result)
+    }
+
+    @Test
+    fun doesNotImportAnythingWhenFileIsNotProperlyFormatted() {
+        val inputStream = FileUtilities.loadResource("bookmarks/bookmarks_invalid.html")
+        val document = Jsoup.parse(inputStream, Charsets.UTF_8.name(), "duckduckgo.com")
+
+        val bookmarks = parser.parseHtml(document)
+
+        Assert.assertTrue(bookmarks.isEmpty())
+    }
+
+    @Test
+    fun canImportFromFirefox() {
+        val inputStream = FileUtilities.loadResource("bookmarks/bookmarks_firefox.html")
+        val document = Jsoup.parse(inputStream, Charsets.UTF_8.name(), "duckduckgo.com")
+
+        val bookmarks = parser.parseHtml(document)
+
+        assertEquals(7, bookmarks.size)
+
+        val firstBookmark = bookmarks.first()
+        assertEquals("https://support.mozilla.org/en-US/products/firefox", firstBookmark.url)
+        assertEquals("Get Help", firstBookmark.title)
+
+        val lastBookmark = bookmarks.last()
+        assertEquals("https://duckduckgo.com/", lastBookmark.url)
+        assertEquals("DuckDuckGo — Privacy, simplified.", lastBookmark.title)
+    }
+
+    @Test
+    fun canImportFromChrome() {
+        val inputStream = FileUtilities.loadResource("bookmarks/bookmarks_chrome.html")
```

Comment:
Do we expect to import favorites from other browsers? If so, does this `html` contain favorites?

Edit: I've just seen we do it for Firefox, right? I assume you can't have favorites on Chrome?

Reply:
We do @cmonfortep. Basically, to us everything is a folder, but if the folder is called DuckDuckGo Favorites then we'll treat it as a favorite. So, if the Chrome file has a folder called DuckDuckGo Favorites we'll treat is such.